data_steward/cdr_cleaner/cleaning_rules/remove_non_matching_participant.py

```diff
@@ -35,214 +45,275 @@
+            column_expr=self.get_missing_criterion(identity_match_fields),
+            num_of_missing=NUM_OF_MISSING_ALL_FIELDS)
+
+        # instantiate the query for identifying the non-match participants in the validation_dataset
+        select_non_match_participants_query = SELECT_NON_MATCH_PARTICIPANTS_QUERY.render(
+            project_id=self.project_id,
+            validation_dataset_id=validation_dataset_id,
+            identity_match_table=identity_match_table,
+            key_fields_criteria=num_of_missing_key_fields,
+            all_fields_criteria=num_of_missing_all_fields,
+            not_validated_participants_expr=NOT_VALIDATED_PARTICIPANTS_TEMPLATE.
+            render(pids=", ".join(str(pid) for pid in pids)) if pids else '')
+
+        return select_non_match_participants_query
+
+    def get_query_specs(self) -> list:
+        """
+        Return a list of dictionary query specifications.
+        The list contains the queries that delete participants and their corresponding data points, 
+        for which the participant_match data is missing and DRC matching algorithm flags it as a no match
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+
+        if self.ehr_dataset_id is None:
+            raise RuntimeError('Required parameter ehr_dataset_id not set.')
+
+        if self.validation_dataset_id is None:
+            raise RuntimeError(
+                'Required parameter validation_dataset_id not set')
+
+        non_matching_person_ids = []
+
+        for hpo_id in readers.get_hpo_site_names():
+            if self.exist_participant_match(self.ehr_dataset_id, hpo_id):
+                not_validated_participants = self.get_not_validated_participants(
+                    self.ehr_dataset_id, hpo_id)
+
+                if not_validated_participants:
+                    LOGGER.info(
+                        f'{hpo_id} submitted the participant_match table, but some of its data is not validated.'
+                    )
+                    non_matching_person_ids.extend(
+                        self.get_non_match_participants(
+                            self.validation_dataset_id,
+                            hpo_id,
+                            pids=not_validated_participants))
+
+                else:
+                    LOGGER.info(
+                        f'{hpo_id} submitted the participant_match table, and all data is validated.'
+                    )
+
+            else:
+                LOGGER.info(f'{hpo_id} is missing the participant_match table.')
+                non_matching_person_ids.extend(
+                    self.get_non_match_participants(self.validation_dataset_id,
+                                                    hpo_id))
+
+        if non_matching_person_ids:
```

Comment:
This looks like logic that is needed for the setup, so would be ideal in setup_rule. I'd say setup_rule should store the results for each hpo in a dictionary, and get_query_specs will use this to determine how many queries to return.